param_sweeps/driver.py

```diff
@@ -104,54 +106,81 @@ def uuid_from_params(params: tuple) -> str:
+
+        return lookup
+
+    def write_files(self, lookup):
+        """Write ui.geoh5 and ui.json files for sweep trials."""
 
         ifile = InputFile.read_ui_json(self.params.worker_uijson)
         with ifile.data["geoh5"].open(mode="r") as workspace:
-            sets = self.params.parameter_sets()
-            iterations = list(itertools.product(*sets.values()))
-            print(
-                f"Running parameter sweep for {len(iterations)} "
-                f"trials of the {ifile.data['title']} driver."
-            )
 
-            param_lookup = {}
-            for count, iteration in enumerate(iterations):
-                param_uuid = SweepDriver.uuid_from_params(iteration)
-                filepath = os.path.join(
-                    os.path.dirname(workspace.h5file), f"{param_uuid}.ui.geoh5"
-                )
-                param_lookup[param_uuid] = dict(zip(sets.keys(), iteration))
-
-                if os.path.exists(filepath):
-                    print(
-                        f"{count}: Skipping trial: {param_uuid}. "
-                        f"Already computed and saved to file."
-                    )
-                    continue
-
-                print(
-                    f"{count}: Running trial: {param_uuid}. "
-                    f"Use lookup.json to map uuid to parameter set."
-                )
-                with Workspace(filepath) as iter_workspace:
-                    ifile.data.update(
-                        dict(param_lookup[param_uuid], **{"geoh5": iter_workspace})
-                    )
-                    objects = [v for v in ifile.data.values() if hasattr(v, "uid")]
-                    for obj in objects:
-                        if not isinstance(obj, Data):
-                            obj.copy(parent=iter_workspace, copy_children=True)
+            for name, trial in lookup.items():
 
-                update_lookup(param_lookup, workspace)
+                status = trial.pop("status")
+                if status == "pending":
+                    filepath = os.path.join(
+                        os.path.dirname(workspace.h5file), f"{name}.ui.geoh5"
+                    )
+                    with Workspace(filepath) as iter_workspace:
+                        ifile.data.update(
+                            dict(lookup[name], **{"geoh5": iter_workspace})
+                        )
+                        objects = [v for v in ifile.data.values() if hasattr(v, "uid")]
+                        for obj in objects:
+                            if not isinstance(obj, Data):
+                                obj.copy(parent=iter_workspace, copy_children=True)
+
+                    ifile.name = f"{name}.ui.json"
+                    ifile.path = os.path.dirname(workspace.h5file)
+                    ifile.write_ui_json()
+                    lookup[name]["status"] = "written"
+                else:
+                    lookup[name]["status"] = status
```

Comment:
Same comment as on GEOPY-621 of geoapps, not sure this logic is optimal

Reply:
In this case, I'm removing the status item so that when I update the ifile.data I don't put a status key in there that doesn't belong

Reply:
Let's chat